pkg/kubelet/dockertools/manager.go

```diff
@@ -1748,16 +1772,69 @@ func (dm *DockerManager) clearReasonCache(pod *api.Pod, container *api.Container
 	dm.reasonCache.Remove(pod.UID, container.Name)
 }
 
+// TODO (random-liu) Move this to container/runtime.go
+
+// SyncAction indicates different kind of actions in SyncPod(). Now there are only actions
+// about start/kill container and setup/teardown network.
+type SyncAction int
+
+const (
+	StartContainer SyncAction = iota
+	KillContainer
+	SetupNetwork
+	TeardownNetwork
+	// Add more later if needed
+)
+
+// SyncResult is the result of sync action.
+type SyncResult struct {
+	// The associated action of the result
+	Action SyncAction
+	// The target of the action, now the target can only be:
+	//  * Container: Target should be container name
+	//  * Network: Target is useless now, we just set it as pod full name now
+	Target interface{}
+	// Brief error reason
+	Error error
```

Comment:
For the current implementation, we can not infer *error* from *action*. For example, for a `StartContainer` action, the error could be pull image error, backoff error, security context error etc.

The error is very short and standard, such as "SetupNetworkError", while the message can be very long and arbitrary.